plugins/vocabbuilder.koplugin/main.lua

```diff
@@ -593,15 +609,61 @@ function VocabItemWidget:initItemWidget()
             width = review_button_width,
             max_width = review_button_width,
             show_parent = self,
-            -- no_focus = true
         }
-
+        local forgot_span_top = IconButton:new{
+            icon = "empty",
+            callback = function() self:onForgot() end,
```

Comment:
This feels a bit ugly - having to use an empty.svg to make an empty IconButton to ensure the padding and have it handle tap in this padding.
Might be ok for a plugin (unless this plugin get so good that it serves as inspiration for future ones :) and I would not like this solution to be advertised and re-used :)

I don't know if we already have that kind of layout with buttons with borders in an horizontal ItemWidget.
I think if I had to do that, I would probably have the Tap handler set on the VocabItemWidget - and ignored/bubbling by its inner widgets. And the tap handler would just do different thing depend on the `x` coordinate along the full VocalItemWidget width.
I can think of this https://github.com/koreader/koreader/blob/2e681353d7d0997d478460254ece5a81ddfee9c6/frontend/ui/widget/pagebrowserwidget.lua#L898 as a unique handler.

Reply:
I agree it's ugly. Let me refactor it to your suggestion. I should have dug a little deeper in onTap's arguments. I'm too used to some rigid static-typed languages and have become dependent on them showing everything on the surface.

Reply:
Done. A lot of if-elses this time.